metadata-integration/java/datahub-schematron/lib/src/test/java/io/datahubproject/schematron/converters/avro/AvroSchemaConverterTest.java

```diff
@@ -0,0 +1,942 @@
+        "string",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(10),
+        "[version=2.0].[type=users_record].[type=address].address.[type=string].country",
+        "string",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(11),
+        "[version=2.0].[type=users_record].[type=address].address.[type=string].postal_code",
+        "string",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(12),
+        "[version=2.0].[type=users_record].[type=address].address.[type=long].created_at",
+        "long(timestamp-micros)",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new TimeType())),
+        "{\"logicalType\":\"timestamp-micros\"}");
+    assertSchemaField(
+        schema.getFields().get(13),
+        "[version=2.0].[type=users_record].[type=contact].contact",
+        "contact",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new RecordType())));
+    assertSchemaField(
+        schema.getFields().get(14),
+        "[version=2.0].[type=users_record].[type=contact].contact.[type=string].email",
+        "string",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(15),
+        "[version=2.0].[type=users_record].[type=contact].contact.[type=string].phone",
+        "string",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(16),
+        "[version=2.0].[type=users_record].[type=long].created_at",
+        "long(timestamp-micros)",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new TimeType())),
+        "{\"logicalType\":\"timestamp-micros\"}");
+    assertSchemaField(
+        schema.getFields().get(17),
+        "[version=2.0].[type=users_record].[type=long].updated_at",
+        "long(timestamp-micros)",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new TimeType())),
+        "{\"logicalType\":\"timestamp-micros\"}");
+    assertSchemaField(
+        schema.getFields().get(18),
+        "[version=2.0].[type=users_record].[type=map].[type=int].props",
+        "int",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
```

Comment:
avro schema

```
    {
      "name": "props",
      "type": [
        "null",
        {
          "type": "map",
          "values": [
            "null",
            "int"
          ]
        }
      ],
      "default": null
    },
```

so this should be

```suggestion
    assertSchemaField(
        schema.getFields().get(18),
        "[version=2.0].[type=users_record].[type=map].[type=int].props",
        "map<string, int>",
        true,
        false,
        new SchemaFieldDataType()
            .setType(
                SchemaFieldDataType.Type.create(
                    new MapType().setKeyType("string").setValueType("int"))));
```

Reply:
you are right, correctly it should be your suggested change